src/main/java/techreborn/blockentity/machine/misc/DrainBlockEntity.java

```diff
@@ -24,17 +24,24 @@
 
 package techreborn.blockentity.machine.misc;
 
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.FluidDrainable;
 import net.minecraft.block.entity.BlockEntityType;
-import net.minecraft.entity.player.PlayerEntity;
-import net.minecraft.inventory.Inventory;
-import reborncore.api.IListInfoProvider;
-import reborncore.api.blockentity.InventoryProvider;
-import reborncore.client.containerBuilder.IContainerProvider;
-import reborncore.client.containerBuilder.builder.BuiltContainer;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.util.math.BlockPos;
 import reborncore.common.blockentity.MachineBaseBlockEntity;
+import reborncore.common.fluid.FluidValue;
+import reborncore.common.fluid.container.FluidInstance;
+import reborncore.common.util.Tank;
 import techreborn.init.TRBlockEntities;
 
-public class DrainBlockEntity extends MachineBaseBlockEntity implements InventoryProvider, IListInfoProvider, IContainerProvider {
+import javax.annotation.Nullable;
+
+public class DrainBlockEntity extends MachineBaseBlockEntity {
+
+	protected Tank internalTank = new Tank("tank", FluidValue.BUCKET, this);
```

Comment:
Internal tank isn't persistent, shouldn't be a problem as it's not meant to be a container.